### PR TITLE
test: Added SirenSubEntity tests

### PR DESCRIPTION
**Pull request by tenbradley**
Context:
[US121367](https://rally1.rallydev.com/#/357252966636d/custom/367300408400?detail=%2Fuserstory%2F441183081456)
The observable subclass SirenSubEntity has tests that are added primarily to test setSirenEntity.
Also added tests to ensure routes are working correctly, which are basically stolen from SirenLink tests, and update SirenLink tests to include some missing coverage

Quality:
- test with siteLord to ensure minor code updates did not affect current Learning Paths
- the tests that are added pass

### CODE REVIEW

File: test/data/observable/entities.js [hunk 1]
```diff
@@ -29,5 +29,38 @@ export const testLinks = [`{
...
 	}]
+  }`, `{
+	"links": [{
+	  "rel": ["bar"],
+	  "href": "http://example.com"
+	}]
+  }`
+];
+
+export const testSubEntitys = [`{
```
Comment (AKobets) [inline test/data/observable/entities.js:40]: typo in `testSubEntitys`

Comment (AKobets) [reply test/data/observable/entities.js:40]: Can we convert `testSubEntitys` to object so test should not rely on the entity item index?
Or export each entity separately with its own name describing its purpose

Comment (tenbradley) [reply test/data/observable/entities.js:40]: Are you saying it should be Entities? I chose to spell it like this to indicate that they are test data for SirenSubEntity and not SirenSubEntities.

Seems reasonable to change it to an object.

Comment (tenbradley) [reply test/data/observable/entities.js]: I have updated the testing data use for Links, SubEntity, and SubEntities to be object and updated the test for all three.

File: test/observable/sirenLink.test.js [hunk 2]
```diff
@@ -138,14 +139,22 @@ describe('sirenLink with observers', () => {
 		});
 
 		it('should update SirenLinks href to be the href from test data', () => {
-			const entity = SirenParse(testLinks[0]);
+			const entity = SirenParse(testLinks.fooRel);
 
 			link1.setSirenEntity(entity);
 			assert.equal(link1.href, 'http://example.com', 'should be href from testLinks[0]');
```
Comment (AKobets) [inline test/observable/sirenLink.test.js:145]: `should be href from testLinks[0]` - this should be updated after changing testLinks from array to object

**Comment by AKobets** [comment]
@tenbradley please check failing tests.

File: test/data/observable/entities.js [hunk 3]
```diff
@@ -1,33 +1,66 @@
...
 	}]
+  }`, barRel: `{
+	"links": [{
+	  "rel": ["bar"],
+	  "href": "http://example.com"
+	}]
+  }`
+};
+
+export const testSubEntitys = { entityWithHref: `{
```
Comment (AKobets) [inline test/data/observable/entities.js:40]: what is going on. it exports both `testSubEntities` and  `testSubEntitys`
Could you please find better names ?

Comment (tenbradley) [reply test/data/observable/entities.js:40]: Switch to componentTests for all three. Now subEntityTests, subEntitiesTests, and linkTests.

**Comment by tenbradley** [comment]
> @tenbradley please check failing tests.

In the process of doing, merging from master broke some stuff as some fields are used differently.

File: test/observable/sirenLink.test.js [hunk 4]
```diff
@@ -138,14 +139,22 @@ describe('sirenLink with observers', () => {
...
 		it('should update SirenLinks href to be the href from test data', () => {
-			const entity = SirenParse(testLinks[0]);
+			const entity = SirenParse(linkTests.fooRel);
 
 			link1.setSirenEntity(entity);
-			assert.equal(link1.href, 'http://example.com', 'should be href from testLinks[0]');
+			assert.equal(link1.href, 'http://example.com', 'should be href from fooRel');
+		});
+
+		it('should not update SirenLinky', () => {
```
Comment (AKobets) [inline test/observable/sirenLink.test.js:148]: typo `SirenLinky`

File: test/observable/sirenLink.test.js [hunk 5]
```diff
@@ -155,10 +164,22 @@ describe('sirenLink with observers', () => {
 
 			link1.setSirenEntity(entity, collection);
 
-			assert.equal(link1.href, 'http://example.com', 'should be href from testLinks[0]');
-			assert.isTrue(link1._observers._observers.has(observer));
+			assert.equal(link1.href, 'http://example.com', 'should be href from linkTests[0]');
```
Comment (AKobets) [inline test/observable/sirenLink.test.js:167]: another place that needs update. Please review all test files and apply similar changes.

**Review by AKobets — DISMISSED** [review]

**Review by AKobets — APPROVED** [review]

**Pull request merged**